### PR TITLE
AnchorFM: use Focused Launch flow

### PR DESCRIPTION
**Pull request by razvanpapadopol**
This PR is similar to https://github.com/Automattic/wp-calypso/pull/48938 so it fixes 391-gh-Automattic/dotcom-manage but this time using Focused Launch flow instead of Gutenboarding pre-launch flow.

### Changes proposed in this Pull Request

* Use Focused Launch ( pbAok1-1IU-p2 ) for AnchorFM sites.
* Display Focused Launch Success View 
  * Show "Your site will be live shortly." message
  * When launch is complete and post is saved, show "Congratulations, your website is now live. We're excited to watch you grow with WordPress." message and action buttons to "Continue editing" or return to My Home.

**Technical notes:**
- `packages/launch/src/hooks/use-on-launch.ts` hook is no longer used in Focused Launch so Free plan doesn't need to be set anymore to prevent redirect to `/checkout` when launching immediately.

### Testing instructions

* Create an AnchorFM site and sandbox it.
* Launch the site from editor.
* Navigate to My Home or continue editing after the site is launched.

### Demo
- **Launch and navigate to My Home**

https://user-images.githubusercontent.com/14192054/110836959-5c9fea00-82a9-11eb-8ab7-5d271b74aa38.mov

- **Launch and continue editing**

https://user-images.githubusercontent.com/14192054/110836953-59a4f980-82a9-11eb-9885-ba5ae2007871.mov

### CODE REVIEW

File: packages/launch/src/focused-launch/success/index.tsx [hunk 1]
```diff
@@ -82,14 +96,16 @@ const Success: React.FunctionComponent = () => {
 	const copyButtonLabelIdle = __( 'Copy Link', __i18n_text_domain__ );
 	const copyButtonLabelActivated = __( 'Copied!', __i18n_text_domain__ );
 
+	const isLaunchComplete = ! isSiteLaunching && isPostSaved;
```
Comment (ciampo) [inline packages/launch/src/focused-launch/success/index.tsx:99]: Note to self: previously, clicking the "Back Home' button in Success view would trigger a browser alert if there were unsaved changes on the post — this is now fixed thank to these changes and the changes highlighted by [this comment](https://github.com/Automattic/wp-calypso/pull/51001/files#r593166500)

File: packages/launch/src/focused-launch/success/index.tsx [hunk 2]
```diff
@@ -35,6 +35,20 @@ const Success: React.FunctionComponent = () => {
...
+	React.useEffect( () => {
+		const asyncSavePost = async () => {
+			if ( ! isPostSaved ) {
+				await savePost();
+				setIsPostSaved( true );
+			}
+		};
+		asyncSavePost();
+	}, [ isPostSaved, savePost ] );
+
```
Comment (ciampo) [inline packages/launch/src/focused-launch/success/index.tsx:51]: Note to self: this change saves the post when the Success view is shown — this avoids the bug described in [this comment](https://github.com/Automattic/wp-calypso/pull/51001/files#r593165382)

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx [hunk 3]
```diff
@@ -58,7 +56,7 @@ registerPlugin( 'a8c-editor-site-launch', {
 						isInIframe: inIframe(),
 					} }
 				>
-					<LaunchModal onClose={ closeSidebar } isLaunchImmediately={ isAnchorFm } />
+					<LaunchModal onClose={ closeSidebar } />
```
Comment (ciampo) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:59]: Should we cleanup the `isLaunchImmediately` prop (and relative logic) from the `LaunchModal` component?

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts [hunk 4]
```diff
@@ -66,6 +66,11 @@ domReady( () => {
 
 			switch ( launchFlow ) {
 				case GUTENBOARDING_LAUNCH_FLOW:
+					// temporary solution, before the backend returns correct launch flow value
+					if ( isAnchorFm ) {
+						dispatch( 'automattic/launch' ).openFocusedLaunch();
+						break;
+					}
```
Comment (ciampo) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Related to comment on https://github.com/Automattic/wp-calypso/pull/51001/files#diff-ed674ad5d1a1fd12d531950e48e3b988ce0cb0593e18a24a206b1da82cca564eR100-R104

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php [hunk 5]
```diff
@@ -97,6 +97,11 @@ function enqueue_launch_flow_script_and_style( $site_launch_options ) {
 			return;
 	}
 
+	$anchor_podcast = $site_launch_options['anchor_podcast'];
+	if ( ! empty( $anchor_podcast ) ) {
+		$script_name = 'focused-launch';
+	}
+
```
Comment (ciampo) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:101]: Why are those changes necessary, if we also apply these changes? https://github.com/Automattic/wp-calypso/pull/51001/files#diff-5ca2166fe3b2f3927d78bb85557a2582921420cc4838e80c276670d324ee504dR69-R73

**Review by ciampo — COMMENTED** [review]
Tests well as per instructions — although I left a few comments that I'd like to clarify before approving

Also, while reviewing I came across these lines:
https://github.com/Automattic/wp-calypso/blob/0db0d6237ae973f15a8ceafa7188d9f2998c785c/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php#L91-L94

They are not changed by this PR, but I wanted to flag it — as agreed, I will open a separate small PR to clean them up

---

Screencast of my test:


https://user-images.githubusercontent.com/1083581/111176442-a2122f00-85a9-11eb-98d5-c75534aa92ee.mp4

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx (hunk 3 above)
Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:59]: No, let's leave it there since it's not directly related to AnchorFM in any way.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php (hunk 5 above)
Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:101]: This code is loading focused launch JS files to the frontend so they can be evaluated at runtime for AnchorFM sites, even if the `$launch-flow` returned from the backend is different than `focused-launch`. Added a comment to cleanup this after we do the backend update.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts (hunk 4 above)
Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Both changes are similar in this way: temporary, to support AnchorFM -> Focused Launch integration before and after we update the launch flow value returned by the backend. 

This one is opening the correct modal but without the other one, Focused Launch JS files are not being loaded so nothing would be rendered.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx (hunk 3 above)
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:59]: Sure thing, thank you for explaining

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php (hunk 5 above)
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:101]: Makes sense!

**Review by ciampo — APPROVED** [review]
LGTM

>  I left a few comments that I'd like to clarify before approving

Thank you for clarifying my questions! I've opened #51113 to track the cleanup changes


> Also, while reviewing I came across these lines:
> https://github.com/Automattic/wp-calypso/blob/0db0d6237ae973f15a8ceafa7188d9f2998c785c/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php#L91-L94
>
> They are not changed by this PR, but I wanted to flag it — as agreed, I will open a separate small PR to clean them up

Opened #51071 to tackle this cleanup separately

**Review by ghost — APPROVED** [review]
Looks good!
